docs/trace_collection.md

```diff
@@ -76,103 +125,230 @@ Send [traces][1] to Datadog from your Android applications with [Datadog's `dd-s
-   ```kotlin
+{{< tabs >}}
+{{% tab "Kotlin" %}}
+```kotlin
    val span = tracer.buildSpan("<SPAN_NAME1>").start()
-   try{
-     val scope = tracer.activateSpan(span)
-     scope.use {
-        // Do something ...
-        doAsyncWork {
-          // Step 2: reactivate the Span in the worker thread
-           val scopeContinuation = tracer.scopeManager().activate(span)
-           scopeContinuation.use {
-              // Do something ...
-           }
+    try {
+        val scope = tracer.activateSpan(span)
+        scope.use {
+            // Do something ...
+            // ...
+            // Start a new Scope
+            val childSpan = tracer.buildSpan("<SPAN_NAME2>").start()
+            try {
+                tracer.activateSpan(childSpan).use {
+                    // Do something ...
+                }
+            } catch(e: Error) {
+                childSpan.error(e)
+            } finally {
+                childSpan.finish()
+            }
         }
-      }   
-   } catch(e: Error) {
-     span.error(e)
-   } finally {
-     span.finish()
-   }
+    } catch(e: Error) {
+        AndroidTracer.logThrowable(span, e);
+    } finally {
+        span.finish()
+    }
+```
+{{% /tab %}}
+{{% tab "Java" %}}
+```java
+    final Span = tracer.buildSpan("<SPAN_NAME1>").start();
+    try {
+        final Scope scope = tracer.activateSpan(span);
+        try {
+            // Do something ...
+            // ...
+            // Start a new Scope
+            final Span childSpan = tracer.buildSpan("<SPAN_NAME2>").start();
+            try {
+                tracer.activateSpan(childSpan).use {
+                // Do something ...
+                }
+            } catch(Error e) {
+                AndroidTracer.logThrowable(childSpan, e);
+            } finally {
+                childSpan.finish();
+            }
+        }
+        finally {
+            scope.close()
+        }
+    } catch(Error e) {
+        span.error(e);
+    } finally {
+        span.finish();
+    }
+```
+{{% /tab %}}
+{{< /tabs >}}
+
+7. To use scopes in asynchronous calls:
 
-   ```  
+{{< tabs >}}
+{{% tab "Kotlin" %}}
+```kotlin
+    val span = tracer.buildSpan("<SPAN_NAME1>").start()
+    try{
+        val scope = tracer.activateSpan(span)
+        scope.use {
+            // Do something ...
+            doAsyncWork {
+                // Step 2: reactivate the Span in the worker thread
+                val scopeContinuation = tracer.scopeManager().activate(span)
+                scopeContinuation.use {
+                    // Do something ...
+                }
+            }
+        }
+    } catch(e: Error) {
+        AndroidTracer.logThrowable(span, e);
+    } finally {
+        span.finish()
+    }
+```
+{{% /tab %}}
+{{% tab "Java" %}}
+```java
+    final Span span = tracer.buildSpan("<SPAN_NAME1>").start();
+    try {
+        final Scope scope = tracer.activateSpan(span);
+        try {
+            // Do something ...
+            doAsyncWork {
```

Comment:
not quite Java :)

Reply:
Yeah...was 100% sure I will miss some, `.md` is too hard to track :)